pkg/agent/baker.go

```diff
@@ -263,7 +263,9 @@ func validateAndSetLinuxNodeBootstrappingConfiguration(config *datamodel.NodeBoo
 	if config.KubeletConfig != nil {
 		kubeletFlags := config.KubeletConfig
 		delete(kubeletFlags, "--dynamic-config-dir")
-		if IsKubernetesVersionGe(config.ContainerService.Properties.OrchestratorProfile.OrchestratorVersion, "1.11.0") {
+		if IsKubernetesVersionGe(config.ContainerService.Properties.OrchestratorProfile.OrchestratorVersion, "1.24.0") {
+			kubeletFlags["--feature-gates"] = remoteFeatureGateString(kubeletFlags["--feature-gates"], "DynamicKubeletConfig")
```

Comment:
is this k8s >= 1.24 check added since dynamic kubelet config flag is removed (and not just set to false) from k8s 1.24?

Reply:
correct https://github.com/kubernetes/kubernetes/pull/106932

Reply:
I think it's actually locked to false anyway, but might as well tidy it up on our side